src/crewai/crew.py

```diff
@@ -623,6 +648,36 @@ def _execute_tasks(
                 )
             self._log_task_start(task, agent_to_use)
 
+            if isinstance(task, ConditionalTask):
+                if futures:
+                    task_outputs.extend(
+                        self._process_async_tasks(futures, was_replayed)
+                    )
+                    futures.clear()
+
+                previous_output = task_outputs[task_index - 1] if task_outputs else None
+                if previous_output is not None and not task.should_execute(
```

Comment:
Is it a safe assumption to say if there are 3 async tasks before a conditional task, that we only want the result from the third async task to be used for our conditional statement?

Reply:
Possibly. users set the order of the agents. So proper crew creations and task orders are needed for this to work properly. I believe we want this to be conditional of a task before it vs all tasks before it (wouldn't really make sense). Future can hold a flag to get more tasks if this is requested but this is an edge case we havent seen yet.